test/acl_auto_configure_test.cpp

```diff
@@ -261,6 +269,27 @@ TEST(auto_configure, simple) {
               (int)m_device_def.autodiscovery_def.accel[0].max_work_group_size);
   CHECK_EQUAL(1, (int)m_device_def.autodiscovery_def.accel[0].is_sycl_compile);
 
+  // Checks for device global entry.
+  CHECK_EQUAL(2, (int)m_device_def.autodiscovery_def.num_device_global);
```

Comment:
Granted the above is not better, but do it the other way around: `2u` instead of casting the value under test.

The rationale is that you know `2u` is a valid unsigned integer, but you don't know what the range of the value under test is in case of a failure. You want to avoid casting a potentially unknown value to a different type, which in the worst case might pass your test despite a wrong result.

Reply:
Then again, why is this needed here and not below, where `size` and `address` are also unsigned?

Reply:
Thanks for all the suggestions,  changes are made accordingly in https://github.com/intel/fpga-runtime-for-opencl/pull/100/commits/b2bdd1201252c7e059c1505345d94fcadecb6aef